swo_aws_extension/flows/steps/assign_mpa.py

```diff
@@ -0,0 +1,100 @@
+import logging
+
+from swo.mpt.client import MPTClient
+from swo.mpt.client.mpt import update_order
+from swo.mpt.extensions.flows.pipeline import Step
+
+from swo_aws_extension.airtable.models import (
+    MPAStatusEnum,
+    get_mpa_view_link,
+)
+from swo_aws_extension.aws.client import AWSClient
+from swo_aws_extension.aws.errors import AWSError
+from swo_aws_extension.constants import PhasesEnum
+from swo_aws_extension.flows.order import (
+    PurchaseContext,
+)
+from swo_aws_extension.notifications import Button, send_error
+from swo_aws_extension.parameters import get_phase, set_mpa_account_id, set_phase
+
+logger = logging.getLogger(__name__)
+
+
+class AssignMPA(Step):
+    def __init__(self, config, role_name):
+        self._config = config
+        self._role_name = role_name
+
+    def __call__(self, client: MPTClient, context: PurchaseContext, next_step):
+        if get_phase(context.order) != PhasesEnum.ASSIGN_MPA:
+            logger.info(
+                f"- Next - Current phase is '{get_phase(context.order)}', "
+                f"skipping as it is not '{PhasesEnum.ASSIGN_MPA}'"
+            )
+            next_step(client, context)
+            return
+
+        if context.mpa_account:
+            logger.info(
+                f"- Next - MPA account {context.mpa_account} "
+                f"already assigned to order {context.order_id}. Continue"
+            )
+        else:
```

Comment:
In order to minimize complexity here you can omit else :-) but you need to add return or next_step() in `if` branch